rpc_jobs/setup_nodepool.yml

```diff
@@ -0,0 +1,109 @@
+- job:
+    name: Setup-Nodepool
+    project-type: pipeline
+    concurrent: false
+    properties:
+      - build-discarder:
+          num-to-keep: 30
+    parameters:
+      - rpc_gating_params
+    triggers:
+      - timed: "@daily"
+    dsl: |
+      library "rpc-gating@${RPC_GATING_BRANCH}"
+      common.shared_slave(){
+        try{
+          common.withRequestedCredentials("cloud_creds, id_rsa_cloud10_jenkins_file"){
+            stage("Prepare Ansible Roles"){
+              // Getting the roles is a bit flaky sometimes, so we
+              // implement a retry to improve the chances of success.
+              retry(3){
+                common.venvGalaxy(
+                  "install",
+                  "-r ${env.WORKSPACE}/rpc-gating/role_requirements.yml",
+                  "-p ${env.WORKSPACE}/roles"
+                )
+                common.venvGalaxy(
+                  "install",
+                  "-r ${env.WORKSPACE}/rpc-gating/nodepool/role_requirements.yml",
+                  "-p ${env.WORKSPACE}/roles"
+                )
+              }
+            } // stage
+
+            stage("Prepare Environment"){
+
+              // write the clouds.yaml file
+              env.OS_CLIENT_CONFIG_FILE = common.writeCloudsCfg(
+                username: env.PUBCLOUD_USERNAME,
+                api_key: env.PUBCLOUD_API_KEY
+              )
+
+              // Tell ansible where to find roles
+              env.ANSIBLE_ROLES_PATH = "${env.WORKSPACE}/roles"
+
+              // Tell ansible where the dynamic inventory is
+              env.ANSIBLE_INVENTORY = "${env.WORKSPACE}/rpc-gating/scripts/ansible_v2_3_2_0_1_contrib_inventory_openstack.py"
+
+              // Tell ansible where the ssh private key is
+              env.ANSIBLE_PRIVATE_KEY_FILE = "${env.JENKINS_SSH_PRIVKEY}"
+
+            } // stage
+
+            stage("Prepare Instances"){
+              dir('rpc-gating/playbooks'){
+
+                // Run the setup playbook
+                common.venvPlaybook(
+                  playbooks: [
+                    "setup_openstack_instances.yml"
+                  ],
+                  args: [
+                    "-v",
+                    "-e '@${env.WORKSPACE}/rpc-gating/nodepool/nodepool-instances.yml'"
+                  ]
+                ) //venvPlaybook
+
+                // We need an inventory refresh, so we execute
+                // this using a fresh shell instead of including
+                // the playbooks in the previous ansible-playbook
+                // play list.
```

Comment:
Note, I'll be adding a refresh inside setup_openstack_instances.yml before we unlock instances (as we need to first ensure the instance exists).  We'll probably want to combine these again to avoid the extra inventory lookup.